src/main.ts

```diff
@@ -345,25 +343,24 @@ function makeLongUtils(options: Options, bytes: ReturnType<typeof makeByteUtils>
   //
   // I.e there is not an import for long that "just works" in both esModuleInterop and
   // not esModuleInterop.
-  const Long = options.esModuleInterop ? imp('Long=long') : imp('Long*long');
+  const LongImp = options.esModuleInterop ? imp('Long=long') : imp('Long*long');
 
   const disclaimer = options.esModuleInterop
     ? ''
     : `
     // If you get a compile-error about 'Constructor<Long> and ... have no overlap',
     // add '--ts_proto_opt=esModuleInterop=true' as a flag when calling 'protoc'.`;
 
-  // Kinda hacky, but we always init long unless in onlyTypes mode. I'd rather do
-  // this more implicitly, like if `Long@long` is imported or something like that.
-  const longInit = options.onlyTypes
-    ? code``
-    : code`
+  const Long = conditionalOutput(
```

Comment:
Ah neat! This is smart. @3846masa do you mind adding an in-source comment right before this:

```
// Instead of exposing `LongImp` directly, let callers think that they are getting the
// `imp(Long)` but really it is that + our long initialization snippet. This means the
// initialization code will only be emitted in files that actually use the Long import.
```

Reply:
Looks good. I'll add it :+1: